### PR TITLE
fixed dashbord error&refactored date hardcoding

### PR DESCRIPTION
**Pull request by uman95**
# Fix network uptime dashboard error

**_WHAT DOES THIS PR DO?_**

* Error handling for newly added devices that has no data
* Removed hard coded dates.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**

**_HOW DO I TEST OUT THIS PR?_**
* cd to network-uptime folder
* create and activate a virtual env
*  then do from terminal `streamlit run network-dashboard.py`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?

[Fix network uptime workflow runtime error](https://github.com/airqo-platform/AirQo-api/pull/799#pullrequestreview-937571231)

**_ARE THERE ANY RELATED PRs?_**

N/A

### CODE REVIEW

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @uman95 
LGTM

File: src/network-uptime/network-dashboard.py [hunk 1]
```diff
@@ -27,11 +29,17 @@
 selected_device = st.sidebar.selectbox('Device name', device_name)
 
 
-@st.cache
+@st.cache(suppress_st_warning=True)
 def load_device_data(device):
-    api = f"{BASE_API_URL}/monitor/devices/uptime?tenant=airqo&startDate=2022-01-01T00:00:00.001Z&endDate=2022-03-25T19:00:00.001Z&device_name=' + device
+    api = f"{BASE_API_URL}/monitor/devices/uptime?tenant=airqo&startDate={START_QUERY_DATE}&endDate={END_QUERY_DATE}&device_name=" + device
```
Comment (NoahNsimbe) [inline src/network-uptime/network-dashboard.py:34]: @uman95, just curious why the start time and end time are environment variables. Won't this limit a user’s ability to customize the date ranges or require updating the .env whenever we want to increase the date range. What about using an interval. some thing like 

```python
interval_in_days = os.getenv('INTERVAL_IN_DAYS', 30)
end_date = datetime.utcnow()
start_date = end_date - timedelta(days=interval_in_days)
START_QUERY_DATE = datetime.strftime(start_date, "%Y-%m-%dT00:00:00Z")
END_QUERY_DATE = datetime.strftime(end_date, "%Y-%m-%dT11:59:59Z")
```
Here we can adjust how the date
Incase you are using months, then you can utilise the function below to get the first and last day of a month

```python
def first_day_of_month(date_time: datetime) -> datetime:
    return datetime(year=date_time.year, month=date_time.month, day=1)

def last_day_of_month(date_time: datetime) -> datetime:
    month = date_time.month
    if month < 12:
        month = month + 1
        next_month = datetime(year=date_time.year, month=month, day=1)
        return next_month - timedelta(days=1)
    else:
        year = date_time.year + 1
        next_month = datetime(year=year, month=1, day=1)
        return next_month - timedelta(days=1)
```

Comment (uman95) [reply src/network-uptime/network-dashboard.py:34]: @NoahNsimbe This PR mainly fix errors from newly added device. The date part is a short term fix (what we currently have actually is the Q1 network uptime, reason for the fixed date range). The whole thing is a work in progress, considering suggestions we got from the last team updates.

Comment (NoahNsimbe) [reply src/network-uptime/network-dashboard.py]: cool @uman95

**Pull request merged**